__tests__/app.test.js

```diff
@@ -1,53 +1,82 @@
+const app = require("../app");
+const db = require("../db/connection");
+const request = require("supertest");
+const testData = require("../db/data/test-data/index.js");
+const seed = require("../db/seeds/seed");
+const {
+  convertTimestampToDate,
+  createRef,
+  formatComments,
+} = require("../db/seeds/utils");
 
 beforeEach(() => seed(testData));
 afterAll(() => db.end());
 
-describe('Northcoders News API', () => {
+describe("Northcoders News API", () => {
+  it("Status: 404 - If the user tries to access an unknown path", () => {
+    return request(app)
+      .get("/api/not-a-route")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("path not found");
+      });
+  });
 
-    it('Status: 404 - If the user tries to access an unknown path', () => {
-        return request(app)
-        .get('/api/not-a-route')
-        .expect(404)
-        .then(({ body }) => {
-            expect(body.msg).toBe('path not found');
-        })
+  describe("GET /api/topics", () => {
+    it("Status: 200 - Should respond with an array of all of the topics.", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ _body: { topicsData } }) => {
+          expect(Array.isArray(topicsData)).toBe(true);
+          expect(topicsData.length).toBe(3);
+          topicsData.forEach((topic) => {
+            const topicKeys = Object.keys(topic);
+            expect(topicKeys).toContain("slug");
+            expect(topicKeys).toContain("description");
+          });
+        });
     });
+  });
 
-    describe('GET /api/topics', () => {
-        it('Status: 200 - Should respond with an array of all of the topics.', () => {
-            return request(app)
-            .get('/api/topics')
-            .expect(200)
-            .then(({_body: {topicsData}}) => {
-                expect(Array.isArray(topicsData)).toBe(true)
-                expect(topicsData.length).toBe(3);
-                topicsData.forEach((topic) => {
-                    const topicKeys = Object.keys(topic);
-                    expect(topicKeys).toContain('slug');
-                    expect(topicKeys).toContain('description');
-                })
+  describe("GET /api/articles/:article_id", () => {
+    it("Status: 200 - Should respond with an article corresponding to the idea given.", () => {
+      return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then(({ _body: { articleData } }) => {
+          const articleKeys = Object.keys(articleData[0]);
+          expect(articleKeys).toHaveLength(7);
+          expect(articleData[0]).toEqual(
+            expect.objectContaining({
+              article_id: expect.any(Number),
+              title: expect.any(String),
+              topic: expect.any(String),
+              author: expect.any(String),
+              body: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
```

Comment:
There's nothing wrong with these tests but they could be tougher. If you are going to articles/1 then you should know what values you are getting back here. For example article_id : 1 etc.

Reply:
Ah I see, that makes sense! Since this is one specific article and it should correspond with the specific values of what it is in the database, I'll get right on it, thank you Christian 😁